modules/commons/src/main/java/org/apache/synapse/commons/json/jsonprocessor/validators/NumericValidator.java

```diff
@@ -0,0 +1,160 @@
+                    throw new ValidatorException("Number " + value + " is greater than the " +
+                            "maximum allowed value. maximumValue constraint in " + inputObject.toString() +
+                            " is violated by the input " + ": " + value);
+                }
+            }
+            // handling exclusive maximum and minimum
+            if (inputObject.has(EXCLUSIVE_MINIMUM)) {
+                String minimumString = inputObject.get(EXCLUSIVE_MINIMUM).getAsString().replaceAll(ValidatorConstants
+                                .REGEX,
+                        "");
+                if (!minimumString.isEmpty() && doubleValue <= DataTypeConverter.convertToDouble(minimumString)) {
+                    throw new ValidatorException("Number " + value + " is less than the " +
+                            "minimum allowed value. ExclusiveMinimum constraint in " + inputObject.toString() +
+                            " is violated by the " + "input : " + value );
+                }
+            }
+            if (inputObject.has(EXCLUSIVE_MAXIMUM)) {
+                String maximumString = inputObject.get(EXCLUSIVE_MAXIMUM).getAsString().replaceAll(ValidatorConstants
+                                .REGEX,
+                        "");
+                if (!maximumString.isEmpty() && doubleValue >= DataTypeConverter.convertToDouble(maximumString)) {
+                    throw new ValidatorException("Number " + value + " is greater than the " +
+                            "maximum allowed value. ExclusiveMaximum constraint in " +
+                            inputObject.toString() + " is violated by the " + "input : " + value);
+                }
+            }
+            // Enum validations
+            if (inputObject.has(ValidatorConstants.ENUM)) {
+                JsonArray enumElements = inputObject.getAsJsonArray(ValidatorConstants.ENUM);
+                if (enumElements.size() > 0 && !enumElements.contains(new JsonPrimitive(doubleValue))) {
+                    throw new ValidatorException("Number \"" + value + "\" not contains any " +
+                            "element from the enum. Input " + value + " not contains any value from the enum in " +
+                            inputObject.toString());
+                }
+            }
+            //Const validation
+            if (inputObject.has(ValidatorConstants.CONST) && !doubleValue.equals(inputObject.getAsJsonPrimitive
+                    (ValidatorConstants.CONST).getAsDouble())) {
+                throw new ValidatorException("Number \"" + value + "\" is not equal to the " +
+                        "const value input " + value + " not contains the const defined in " + inputObject.toString());
+            }
+            // convert to integer of give value is a float
+            if (type != null && type.equals(INTEGER_STRING)) {
```

Comment:
switch this to INTEGER_STRING.equals(type) so we don't need type null check

Reply:
Fixed